enclave-runtime/src/tls_ra/tls_ra_client.rs

```diff
@@ -144,7 +148,7 @@ where
 	/// Read all bytes into a buffer of given length.
 	fn read_until(&mut self, length: usize) -> EnclaveResult<Vec<u8>> {
 		let mut bytes = vec![0u8; length];
-		self.tls_stream.read(&mut bytes)?;
+		self.tls_stream.read_exact(&mut bytes)?;
```

Comment:
Thanks a lot for fixing my bug. Great finding!

Reply:
It was a tricky one, took me a while to understand what was going on 😄